packages/react-native-editor/src/setup-locale.js

```diff
@@ -0,0 +1,55 @@
+/**
+ * WordPress dependencies
+ */
+import { setLocaleData } from '@wordpress/i18n';
+
+/**
+ * Setup locale data for default domain and plugins.
+ *
+ * @typedef {Object} PluginTranslation
+ * @property {string}              domain                Domain of the plugin.
+ * @property {Function}            getTranslation        Function for retrieving translations for a locale.
+ *
+ * @param    {string}              locale                Locale value.
+ * @param    {Object}              extraTranslations     Extra translations to be included.
+ * @param    {Function}            getDefaultTranslation Default domain's function for retrieving translations for a locale.
+ * @param    {PluginTranslation[]} pluginTranslations    Array with plugin translations.
+ */
+export default (
+	locale,
+	extraTranslations,
+	getDefaultTranslation,
+	pluginTranslations
+) => {
+	const setDomainLocaleData = ( getTranslation, domain ) => {
+		let translations = getTranslation( locale );
+		if ( locale && ! translations ) {
+			// Try stripping out the regional
+			locale = locale.replace( /[-_][A-Za-z]+$/, '' );
+			translations = getTranslation( locale );
+		}
+		const allTranslations = Object.assign(
+			{},
+			translations,
+			extraTranslations
+		);
+		// eslint-disable-next-line no-console
+		console.log(
+			domain ? `${ domain } - locale` : 'locale',
+			locale,
+			allTranslations
+		);
+		// Only change the locale if it's supported by gutenberg
+		if ( translations || extraTranslations ) {
+			setLocaleData( allTranslations, domain );
+		}
+	};
+
+	// Set up default locale data (Gutenberg)
+	setDomainLocaleData( getDefaultTranslation );
+
+	// Set up plugin translations
+	pluginTranslations.forEach( ( { domain, getTranslation } ) =>
+		setDomainLocaleData( getTranslation, domain )
+	);
```

Comment:
This part will allow adding translations of other plugins.

Reply:
Is it accurate to say this module is unused today? If so, what is an explicit example usage we have planned?

Reply:
> Is it accurate to say this module is unused today? If so, what is an explicit example usage we have planned?

Yes, that's right, it's not being used yet. Once [this PR](https://github.com/wordpress-mobile/gutenberg-mobile/pull/4359) gets merged, I'll update the GB-mobile PR to use this part.

Here is a patch diff related to the [GB-mobile PR](https://github.com/wordpress-mobile/gutenberg-mobile/pull/4332) as an example:
```diff
diff --git a/src/index.js b/src/index.js
index c1036474850da443cf10d1e6c9b40a9cfb7ea90e..f825dd085cbf7ac9d89f0c50fb165aa60f128b72 100644
--- a/src/index.js
+++ b/src/index.js
@@ -3,6 +3,23 @@
  */
 import { registerGutenberg } from '@wordpress/react-native-editor';
 
+/**
+ * Internal dependencies
+ */
+import { getTranslation as getJetpackTranslation } from './i18n-cache/jetpack';
+import { getTranslation as getLayoutGridTranslation } from './i18n-cache/layout-grid';
+
+const pluginTranslations = [
+	{
+		domain: 'jetpack',
+		getTranslation: getJetpackTranslation,
+	},
+	{
+		domain: 'layout-grid',
+		getTranslation: getLayoutGridTranslation,
+	},
+];
+
 registerGutenberg( {
 	beforeInitCallback: () => {
 		// We have to lazy import the setup code to prevent executing any code located
@@ -10,4 +27,5 @@ registerGutenberg( {
 		const { default: gutenbergMobileSetup } = require( './setup' );
 		gutenbergMobileSetup();
 	},
+	pluginTranslations,
 } );
```

Reply:
Now that the [mentioned PR](https://github.com/wordpress-mobile/gutenberg-mobile/pull/4359) is merged, you can see an example of the plugin translations usage in the GB-mobile PR ([reference](https://github.com/wordpress-mobile/gutenberg-mobile/blob/504270493676fe9c2a219519bb70a8d76106dee1/src/index.js)).